simple_database/main.py

```diff
@@ -1,44 +1,133 @@
 import os
 from datetime import date
+import json
 
 from simple_database.exceptions import ValidationError
 from simple_database.config import BASE_DB_FILE_PATH
 
+# Handle date objects not being json friendly
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, date):
+            return o.isoformat()
+        return json.JSONEncoder.default(self, o)
+
+class RowEntry(object):
+    def __init__(self, **kwargs):
+        for k in kwargs.keys():
+            setattr(self, k, kwargs[k])
 
 class Table(object):
+    
+    VALIDATION_ERROR_FORMAT = 'Invalid type of field "{}": Given "{}", expected "{}"'
 
     def __init__(self, db, name, columns=None):
-        pass
+        self.db = db
+        self.name = name
+        self.columns = columns
+        self.path = BASE_DB_FILE_PATH + db.name + os.sep + name
+        self._read()
 
     def insert(self, *args):
-        pass
-
-    def query(self, **kwargs):
-        pass
+        row_data = {}
+        # verify no. input fields matches no. colums
+        if not len(args) == len(self.columns):
+            raise ValidationError('Invalid amount of field')
+        for i, col in enumerate(self.columns):
+            field_name = col['name']
+            field_type = col['type']
+            input_type = type(args[i]).__name__
+            if field_type == input_type:
+                row_data[field_name] = args[i]
```

Comment:
Very good idea to add all values into a dict which will index them by column name.